flycheck.el

```diff
@@ -9480,6 +9480,26 @@ or if the current buffer has no file name."
   (and buffer-file-name
        (locate-dominating-file buffer-file-name "Cargo.toml")))
 
+(defun flycheck-rust-cargo-metadata ()
+  "Exec 'cargo metadata' and return the result as parsed JSON object.
+
+Return nil if the command failed or if a parse error occured."
```

Comment:
Do we really want to fail silently for failed calls to `cargo metadata` and for parse errors? (IIUC this is for backwards compatibility, but I'm afraid it'll create harder-to-debug bugs if things change unexpectedly)

Reply:
I think it's only because 'cargo' 's bug if parse error occurs, so we can allow error here, but I'm not sure.